plugins/plugin-maven/che-plugin-maven-ide/src/main/java/org/eclipse/che/plugin/maven/client/actions/GetEffectivePomAction.java

```diff
@@ -77,49 +80,49 @@ public void updateInPerspective(@NotNull ActionEvent event) {
       return;
     }
 
-    final Optional<Project> project = resource.getRelatedProject();
-    if (!project.isPresent()) {
+    final Project project = resource.getProject();
+    if (project == null) {
       event.getPresentation().setEnabledAndVisible(false);
       return;
     }
 
-    event.getPresentation().setEnabledAndVisible(project.get().isTypeOf(MAVEN_ID));
+    event.getPresentation().setEnabledAndVisible(project.isTypeOf(MAVEN_ID));
   }
 
   @Override
   public void actionPerformed(ActionEvent e) {
-    final Resource[] resources = appContext.getResources();
-
-    checkState(resources != null && resources.length == 1);
-
-    final Project project = resources[0].getRelatedProject().get();
+    final Resource resource = appContext.getResource();
+    checkState(resource != null);
 
+    final Project project = resource.getProject();
     checkState(MAVEN_ID.equals(project.getType()));
 
-    mavenServerServiceClient
-        .getEffectivePom(project.getLocation().toString())
+    final String pathToProjectsDir = "/projects";
```

Comment:
There is no knowledge of the "projects" folder in the front end. The conversion of a workspace path to a file system location should be done in the wsagent, not here. See class "LanguageServiceUtils".

Reply:
Yes, this is my mistake. Thank you.